### PR TITLE
Add likeable to config for Chatbot

### PR DESCRIPTION
**Pull request by freddyaboulton**
## Description

Repro in issue.

We added some under-the-hood logic to Blocks to set `_selectable` in the config without it being present in the init signature. I don't like treating some props as privileged though. Doesn't scale for custom component authors who want to set event-based props in their custom component. So I'm adding likeable to the config.

Closes: #6230 

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: test/test_components.py [hunk 1]
```diff
@@ -1895,6 +1895,16 @@ def test_component_functions(self):
             "_selectable": False,
         }
 
+    def test_chatbot_selectable_in_config(self):
```
Comment (abidlabs) [inline test/test_components.py:1898]:
```suggestion
    def test_chatbot_likeable_in_config(self):
```

Comment (abidlabs) [reply test/test_components.py:1898]: Maybe we can also test to ensure that likable is _not_ present if no `.like` event has been set

**Comment by abidlabs** [comment]
> I don't like treating some props as privileged though. Doesn't scale for custom component authors who want to set event-based props in their custom component. So I'm adding likeable to the config.

Makes sense. Let's also do the same for `_selectable` then for consistency across components. We'll add it back as a param to all of the components and do it in a backwards compatible way to support `gr.load()` components with the `_selectable` key. I can take care of this in a separate PR.

**Comment by abidlabs** [comment]
Oh tested, works well. LGTM!

File: test/test_components.py (hunk 1 above)
Comment (freddyaboulton) [reply test/test_components.py:1898]: Yea good idea!

**Comment by freddyaboulton** [comment]
Thanks @abidlabs !

**Pull request merged**